### DISCUSSION TITLE
Automated conversion from HF model definition?

### DISCUSSION THREAD

**Discussion by am17an** [ggml-org/llama.cpp#16750] [Ideas]
Not sure if this has been discussed before, but I'm trying to understand how feasible this is. As such this would provide support for "free" for models if all the operations are already supported in ggml/llama.cpp.

From my understanding, everything finally boils down a compute graph with different operations and input/output tensors. If some how we are able take the compute graph and convert it automatically then it would be a very useful feature. Obviously there are other complexities with tokenizers, chat templates etc. but getting logits to match would be the logical first step

**Comment by nisparks** [comment]
I agree, and I'm working on that myself. One issue for full creation right now is that the GGUF is missing metadata for hidden activation functions. Working on a change there as well.
